### PR TITLE
rgw: add support for account and container quotas of Swift API

### PR DESCRIPTION
**Pull request by rzarzynski**
This pull request brings support for account as well as container quota of Swift API. The first one isn't currently operational because RadosGW doesn't have the *reseller admin* concept yet. It's essential because only administrators should be able to modify account quotas - there is no business in having that kind of quota if we allow account owner to change it. This is stays in contrast to bucket quota that may be altered by anyone with write access to a given bucket.

Derogates: #8947.

Depends on: *reseller admin* feature of #8657.

Includes commits from: #9002.

CC: @yehudasa, @mattbenjamin.

### CODE REVIEW

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -1876,6 +1878,64 @@ static void populate_with_generic_attrs(const req_state * const s,
...
+      return -EINVAL;
+    }
+    add_attrs.erase(iter);
+    extracted = true;
+  }
+
+  /* Put new limit on bucket (container) size. */
+  iter = add_attrs.find(RGW_ATTR_QUOTA_MSIZE);
+  if (iter != add_attrs.end()) {
+    const int64_t max_size =
```
Comment (oritwas) [inline src/rgw/rgw_op.cc]: why do you need the temp variable?

Comment (rzarzynski) [inline src/rgw/rgw_op.cc]: Right, it's completely unnecessary. Thanks for pointing that.
I will push the amended version soon.

**Comment by rzarzynski** [comment]
@oritwas: I've pushed a slightly amended version. The temp variable has been removed, debugs improved (a new one with priority `20` in the end of `RGWQuotaHandlerImpl::check_quota` to signalize positive a quota verification) and rounding in `RGWQuotaInfoDefApplier::is_size_exceeded` fixed

**Comment by rzarzynski** [comment]
@oritwas: rebased due to a merge conflict.

**Comment by oritwas** [comment]
@rzarzynski , can you rebase?

**Comment by rzarzynski** [comment]
@oritwas: rebased. The previous version [is available](https://github.com/rzarzynski/ceph/tree/wip-rgw-swift-quota-old-2) as well.

**Pull request merged**